### DISCUSSION TITLE
how to integrate "daily forecast"

### DISCUSSION THREAD

**Discussion by natnatd** [cytech/Home-Assistant-wundergroundpws#88] [General]
Hi,
thanks a lot, i succeed to add my PWS to HA even if i'm a newbie !
i just don't understand how to put the "daily_forecast:" part,

i tried in my sensors.yaml folder, like this : 

- platform: wundergroundpws
  api_key: XXXXXXXX
  pws_id: XXXXXX
  numeric_precision: decimal
  lang: fr-FR
  monitored_conditions:
      - solarRadiation
      - winddir
      - humidity
      
  daily_forecast:
    name: Daily Forecast
    entities:
      - sensor.pws_weather_1d
      - sensor.pws_weather_1n
      - sensor.pws_weather_2d
      - sensor.pws_weather_2n
      - sensor.pws_weather_3d
      - sensor.pws_weather_3n
      - sensor.pws_weather_4d
      - sensor.pws_weather_4n

but :
Invalid config for [sensor.wundergroundpws]: [daily_forecast] is an invalid option for [sensor.wundergroundpws]. Check: sensor.wundergroundpws->daily_forecast. (See ?, line ?).

it works with the first part but not with the daily forecast part...

really new in yaml so i'm not sure about the spaces or the tabs, perhaps that's the problem ?

i tried different thinks but still doesn't work...

be really happy if you can help me :)

**Comment by cytech** [comment]
formatting is everything in yaml........

    sensor:
      - platform: wunderground
        api_key: YOUR_API_KEY
        pws_id: YOUR_STATION_ID
        monitored_conditions:
          - weather_1d
          - weather_1n
          - weather_2d
          - weather_2n
          - weather_3d
          - weather_3n
          - weather_4d
          - weather_4n

      daily_forecast:
        name: Daily Forecast
        entities:
          - sensor.pws_weather_1d
          - sensor.pws_weather_1n
          - sensor.pws_weather_2d
          - sensor.pws_weather_2n
          - sensor.pws_weather_3d
          - sensor.pws_weather_3n
          - sensor.pws_weather_4d
          - sensor.pws_weather_4n

**Comment by natnatd** [comment]
you're so right, really triing to learn !!

here is the code in my /config/sensors.yaml

<img width="464" alt="Capture d’écran 2022-02-23 à 21 43 51" src="https://user-images.githubusercontent.com/99091140/155405143-fb01c693-04d8-471f-ab8b-840f44cfd044.png">

here is the error report now : 
Error loading /config/configuration.yaml: while parsing a block collection
in "/config/sensors.yaml", line 1, column 1
expected <block end>, but found '?'
in "/config/sensors.yaml", line 31, column 1

**Comment by cytech** [comment]
its your formatting
the whole "daily forecast" block has to be indented even with the "monitored conditions"
see examples in the readme
https://github.com/cytech/Home-Assistant-wundergroundpws

**Reply by natnatd** [reply]
i really tried and read the exemple but still doesn't work,
i indented with the same spaces as "monitored conditions"

<img width="461" alt="Capture d’écran 2022-02-23 à 21 51 57" src="https://user-images.githubusercontent.com/99091140/155406965-ee87fa00-8341-463c-a611-f17f1786c4c8.png">


but here is the error now :
Invalid config for [sensor.wundergroundpws]: [daily_forecast] is an invalid option for [sensor.wundergroundpws]. Check: sensor.wundergroundpws->daily_forecast. (See ?, line ?).

**Comment by cytech** [comment]
moving to discussions

**Reply by natnatd** [reply]
sorry not to be in the good section !

**Reply by cytech** [reply]
all indents should be 2 spaces
it looks like you have 4 spaces under the monitored conditions items

**Reply by natnatd** [reply]
i tried with 2 spaces and still have the error message : 
Invalid config for [sensor.wundergroundpws]: [daily_forecast] is an invalid option for [sensor.wundergroundpws]. Check: sensor.wundergroundpws->daily_forecast. (See ?, line ?).

it's strange because the monitored conditions works even with 4 spaces when daily_forecast part is delete...

**Reply by cytech** [reply]
im not sure
ill try and look at it this week

**Reply by natnatd** [reply]
thanks a lot for your answers and your time,
not a big issue if it doesn't work here ;)

**Comment by cytech** [comment]
sorry
that "daily forecast" should not be part of the configuration.
do not use it.
it was a bit left over in the readme for defining the widget outside of the HA interface.
i have updated the readme.

**Reply by natnatd** [reply]
ok thanks for your reply,
i thought it was to define a weather entity to the sensors, to have forecast prevision :)

**Reply by cytech** [reply]
no, the sensor is already defined in the monitored_conditions so you can access it in HA.
that snippet was an example of the old widget definition

**Comment by cytech** [comment]
closed as outdated

**Comment by cytech** [comment]
closed as outdated
